Update the Operator Install annd Kiali Uninstall sections of getting started (#139)
kiali/kiali.io: content/documentation/getting-started/index.adoc

If you prefer to use the latest Kiali version then do not install Kiali during the Istio or Maistra installation procedure. Complete the Istio or Maistra installation and then install Kiali using the steps below. Note that the Kiali installation script documented below will ask to delete an existing Kiali installation, if found.

=== Install Kiali Operator

If you did not install Kiali with Istio or Maistra, or want to replace the installed version of Kiali, it is done using the Kiali Operator. The Kiali Operator is an implementation of a link:https://coreos.com/operators/[Kubernetes Operator]. It provides a method of packaging, deploying, and managing Kiali.

All of the installation options below:

* Ask to delete an existing Kiali Operator, or Kiali installation, when appropriate.
* Prompt for the authentication strategy to use and/or additional information needed for login.
** See link:#_login_options[Login Options] for more on the `login`, `anonymous` and `openshift` authentication strategies.

> 

Reviewer: the above is confusing. It doesn't flow. Not sure what its trying to convey.

Author: I will replace the bullets with prose and see if you like it better.

==== Default Installation

> This option installs the Kiali Operator and the Kiali CR (Custom Resource), with default options.  Kiali will be able to access existing namespaces, but will require a CR update to access future namespaces (for more, see link:#_namespace_management[Namespace Management]. Run the install script via this command:

Reviewer: Missing parens
```suggestion
This option installs the Kiali Operator and the Kiali CR (Custom Resource), with default options.  Kiali will be able to access existing namespaces, but will require a CR update to access future namespaces (for more, see link:#_namespace_management[Namespace Management]). Run the install script via this command:
```

Author: I'll get rid of the parens completely.

[source,bash]
----
bash <(curl -L https://git.io/getLatestKialiOperator)
----

==== Demo/Dev Installation

> Like the Default Installation but Kiali will be able to access existing namespaces and future namespaces. Because this grants cluster roles it is not recommended for production. Run the install script via this command:

Reviewer: ```suggestion
Like the Default Installation but Kiali will be able to access existing namespaces and future namespaces. Because this grants special cluster role permissions, it is not recommended for production. Run the install script via this command:
```

Author: +1

[source,bash]
----

> bash <(curl -L https://git.io/getLatestKialiOperator) --accessible-namespaces

Reviewer: ```suggestion
bash <(curl -L https://git.io/getLatestKialiOperator) --accessible-namespaces '**'
```

Author: oops, that's important

----

==== Custom Installation

This option installs the Kiali Operator and a Kiali CR (Custom Resource) with a custom configuration. To generate the CR file it is recommended to copy the fully documented link:https://github.com/kiali/kiali/blob/master/operator/deploy/kiali/kiali_cr.yaml[example Kiali CR yaml file].  Edit the file, being careful to maintain proper formatting, and save it to a local file (e.g. my-kiali-cr.yaml).

icon:lightbulb[size=1x]{nbsp} It is important to understand the `deployment:accessible_namespaces` setting in the CR.  See link:#_accessible_namespaces[Accessible Namespaces] for more information.

Run the install script via this command (replacing the CR filename as needed):

[source,bash]
----
bash <(curl -L https://git.io/getLatestKialiOperator) --kiali-cr my-kiali-cr.yaml
----

==== Post Install

> After installation is complete using one of the above options you are free to update your Kiali CR.  The Kiali Operator will automatically apply any updates.  To edit the Kiali CR run the command (note: the default kiali CR name is `kiali`):

Reviewer: Maintain consistent capitalization. Everywhere else its called `Kiali CR` (not kiali CR)
```suggestion
After installation is complete using one of the above options you are free to update your Kiali CR.  The Kiali Operator will automatically apply any updates.  To edit the Kiali CR run the command (note: the default Kiali CR name is `kiali`):
```

Author: +1 fixed in multiple places

[source,bash]
----

> oc edit kiali <kiali CR name> -n kiali-operator

Reviewer: ```suggestion
oc edit kiali <Kiali CR name> -n kiali-operator
```

----

Or to create or update using a local file (here `my-kiali-cr.yaml`):

[source,bash]
----
oc apply -f my-kiali-cr.yaml -n kiali-operator
----

===== Kiali Operator Install Script

It is not usually necessary to explicitly execute the link:https://github.com/kiali/kiali/blob/master/operator/deploy/deploy-kiali-operator.sh[Kiali Operator install script] but it does offer a variety of options.  To learn more about the script, and the available options, run the command:

[source,bash]
----

> ./operator/deploy/deploy-kiali-operator.sh --help

Reviewer: It is still possible to run this script without downloading it.  You can still do:

```
bash <(curl -L https://git.io/getLatestKialiOperator) --help
```

Not sure if it matters, just fyi

Author: I'll change it, no reason to show a file location when not needed.

Then the URL is `https://localhost:20001/kiali`.

== Uninstall

> To uninstall Kiali is simple - just delete the Kiali CR. To trigger the Kiali Operator to uninstall Kiali run the command (note: the default kiali CR name is `kiali`):

Reviewer: ```suggestion
To uninstall Kiali is simple - just delete the Kiali CR. To trigger the Kiali Operator to uninstall Kiali run the command (note: the default Kiali CR name is `kiali`):
```

[source,bash]
----

> oc delete kiali <kiali CR name> -n kiali-operator

Reviewer: ```suggestion
oc delete kiali <Kiali CR name> -n kiali-operator
```

----

At this point, you have no Kiali installed, but you still have the Kiali Operator running. You could create another Kiali CR (with potentially different configuration settings) to install a new Kiali instance.

To uninstall *everything* related to Kiali run the command:

[source,bash]
----

> bash <(curl -L https://git.io/getLatestKialiOperator) --uninstall-mode

Reviewer: ```suggestion
bash <(curl -L https://git.io/getLatestKialiOperator) --uninstall-mode true
```

Author: +1
